core/src/invalid_fee_payer_filter.rs

```diff
@@ -0,0 +1,108 @@
+//! Manages a set of recent invalid fee payers.
+//! This is used to filter out transactions that are unlikely to be included
+//! in any block, much earlier in the transaction pipeline.
+//!
+
+use {
+    dashmap::DashSet,
+    solana_sdk::pubkey::Pubkey,
+    std::{
+        sync::{
+            atomic::{AtomicBool, AtomicUsize, Ordering},
+            Arc,
+        },
+        thread::{sleep, Builder, JoinHandle},
+        time::Duration,
+    },
+};
+
+#[derive(Default)]
+pub struct InvalidFeePayerFilter {
+    recent_invalid_fee_payers: DashSet<Pubkey>,
+    reject_count: Arc<AtomicUsize>,
+}
+
+impl InvalidFeePayerFilter {
+    /// Reset the filter - this should be called periodically.
+    pub fn reset(&self) {
+        self.recent_invalid_fee_payers.clear();
+    }
+
+    /// Add a pubkey to the filter.
+    pub fn add(&self, pubkey: Pubkey) {
+        self.recent_invalid_fee_payers.insert(pubkey);
+    }
+
+    /// Check if a pubkey is in the filter.
+    pub fn should_reject(&self, pubkey: &Pubkey) -> bool {
+        let should_reject = self.recent_invalid_fee_payers.contains(pubkey);
+        if should_reject {
+            self.reject_count.fetch_add(1, Ordering::Relaxed);
+        }
+        should_reject
+    }
+}
+
+/// Simple wrapper thread that periodically resets the filter.
+pub struct InvalidFeePayerFilterThread {
```

Comment:
hmm re-thinking after I posted the PR. I probably don't really need a separate thread for this at all.

1. Just add an atomic interval on the `InvalidFeePayerFilter`
2. Banking threads just call some fn that resets & reports if atomic interval should report